quint/src/quintVerifier.ts

```diff
@@ -182,7 +184,76 @@ const grpcStubOptions = {
   oneofs: true,
 }
 
-function loadGrpcClient(dist: ApalacheDist): VerifyResult<AsyncCmdExecutor> {
+async function loadProtoDefViaReflection(): Promise<VerifyResult<ProtoPackageDefinition>> {
+  type ServerReflectionResponseSuccess = {
+    file_descriptor_response: {
+      file_descriptor_proto: Buffer[]
+    }
+  }
+  type ServerReflectionResponseFailure = {
+    error_response: {
+      error_code: number
+      error_message: string
+    }
+  }
+  type ServerReflectionResponse = ServerReflectionResponseSuccess | ServerReflectionResponseFailure
+  type ServerReflectionService = {
+    new (url: string, creds: any): ServerReflectionService
+    ServerReflectionInfo: () => any
+  }
+  type ServerReflectionPkg = {
+    grpc: { reflection: { v1alpha: { ServerReflection: ServerReflectionService } } }
+  }
```

Comment:
Perhaps this coming from too much of a static mindset, but it makes me feel it a bit nervous to have types generated dynamically in a function body! XD

Is this a normal typescript idiom? Any reason we can't define these outside of the function?

Reply:
We can definitely put it outside. I kept them in the function as I wanted to keep them in limited scope. But, it uses the same style used for CmdExecutor.

Reply:
Thanks for explaining! Thomas and Gabriela also explained that this is just a way of controlling the scope of the type aliases. I'd not but it inside a function myself, but won't block on it, and iiuc, Thomas seems to like this approach :)